src/ui/units/ql/containers/PaneMain/ScreenEditor/TabParams/TabParams.tsx

```diff
@@ -273,64 +316,83 @@ class TabParams extends React.PureComponent<TabParamsProps, TabParamsState> {
+                                        to: toDate,
+                                    };
+
+                                    this.onEditParam({param: updatedParam, paramIndex});
+                                }
+                            },
+                        });
                     }}
-                    hasCalendarIcon={false}
-                    allowNullableValues={false}
-                    hasClear={false}
-                    withTime={param.type === QLParamType.DatetimeInterval}
-                    timeZone={DEFAULT_TIMEZONE}
-                />
+                    key="button-open-param-dialog"
+                    className={b('default-value-text')}
+                    qa="open-param-dialog-btn"
+                >
+                    {param.defaultValue
+                        ? `${formattedFrom} — ${formattedTo}`
+                        : i18n('sql', 'label_placeholder-default-value')}
+                </Button>
             );
         } else if (paramIsDate && typeof param.defaultValue === 'string') {
+            const formattedDate = resolveAndFormatDate(
+                param.defaultValue,
+                param.type as QLParamType,
+            );
+
             return (
-                <SimpleDatepicker
-                    wrapClassName={b('default-value-date')}
-                    date={param.defaultValue}
-                    withTime={param.type === QLParamType.Datetime}
-                    allowRelative={true}
-                    hasClear={false}
-                    onUpdate={({date}: {date: string | null}) => {
-                        const newParam = {...param};
-
-                        if (date !== null) {
-                            newParam.defaultValue = date;
-                        }
-
-                        this.onEditParam({param: newParam, paramIndex});
+                <Button
+                    view="outlined"
+                    size="m"
+                    onClick={() => {
+                        this.onClickButtonEditParamValue({
+                            value: param.defaultValue as string,
+                            type: param.type as QLParamType,
+                            onApply: ({value}) => {
+                                const newParam = {...param};
+
+                                if (value !== null) {
+                                    newParam.defaultValue = value;
+                                }
+
+                                this.onEditParam({param: newParam, paramIndex});
+                            },
+                        });
                     }}
-                />
+                    key="button-add-param-value"
+                    className={b('default-value-text')}
+                    qa="open-param-dialog"
+                >
+                    {param.defaultValue
+                        ? formattedDate
+                        : i18n('sql', 'label_placeholder-default-value')}
+                </Button>
```

Comment:
Is it possible to take out some logical pieces as separate components/utilities? The file was already not small, but now it has become even larger